package.json

```diff
@@ -81,6 +81,7 @@
     "jest-plugin-context": "^2.9.0",
     "jest-runner-tsc": "^1.6.0",
     "jest-sonar-reporter": "^2.0.0",
+    "prettier": "2.0.2",
```

Comment:
I am confused why we have prettier changes in this PR. I don't think the ticket is mentioning it. Also, will these formatting changes impact the existing linting (and CI)?. It might be worth adding a command or a to a hook for it. Why still using the .editorconfig ? Prettier should format the code for us before committing. I can see a max line of length of 180 whereas prettier has 98 for example.

Reply:
Prettier was added recently primarily for code formatting as it is better at it than ESLint in regards to code formatting. . It was actually planned to go with the ADR changes but it is added over here as well. ADR would go into Develop first so this would end up conflicting. We can remove it from this branch and keep it with ADR.

Reply:
@JonLazariniDD having prettier in this ticket does not cause much of an issue at this stage. Yes it was added and needed as part of the huge ADR refactor ticket. The VTM app currently lacks formatting and proper code structuring. The developer added it here to start conforming to the same code layout and structure before we eventually merge ADR to develop. Hence once ADR goes to develop this wont be a problem. So I suggest we leave it for now now. With respect to .editorConfig this is a default with most Angular setup project. We are using Prettier going forward and it supersedes .editorConfig

Reply:
ok, are we sure it is not going to cause issues with SonarQube? I tried to check the log and seen that the CI env does not have the correct version of Ts installed 2.9.2 whereas the CI is looking for 3.x. which is a different issue.
```
ERROR: You are using version of TypeScript 2.9.2 which is not supported; supported versions >=3.2.1
```
I can see that sonarQube is expecting a linting report to be analysed
https://github.com/dvsa/cvs-app-vtm/blob/develop/sonar-project.properties#L18
but it seems we are never generating that so I assume the CI won't pick this up?
I don't have access to SonarQube yet to verify the generated report unfortunately.

Reply:
@JonLazariniDD you hit the nail right on the head. From my point of view and the analysis so far, we are in somewhat of a virgin island. So as much of the all the necessary good stuff and best practices can be pushed through for now. The VTM app needs a lot of TLC. With respect to typescript version - this is another deficiency with the VTM app. Ideally we should be advising for an upgrade to ng9

Reply:
Closing it because is outdated